lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/YarnContainerManager.java

```diff
@@ -535,13 +538,38 @@ private void handleNewContainer(final Container container) {
       LOG.log(Level.FINEST, "Allocated Container: memory = {0}, core number = {1}",
           new Object[] {container.getResource().getMemory(), container.getResource().getVirtualCores()});
 
+      // TODO[JIRA REEF-1866]: update this part when YARN supports a method to retrieve
+      // the node label expression of a container.
+      final YarnClient client = YarnClient.createYarnClient();
+      client.init(new YarnConfiguration());
+      client.start();
+
+      String nodeLabelExpressionsFromNode = null;
+
+      try {
+        if (client.getNodeToLabels().get(container.getNodeId()) != null) {
+          for (final String nodeLabelExpression: client.getNodeToLabels().get(container.getNodeId())) {
+            nodeLabelExpressionsFromNode = nodeLabelExpression;
+          }
+        }
+      } catch (final YarnException | IOException e) {
+        LOG.log(Level.WARNING, "Error getting label.", e);
```

Comment:
Please be more expressive in the log message here. What happens when we ignore these exceptions? Will the scheduling constraint be ignored? If so, print which constraint got ignored as part of the log.